superset/assets/src/SqlLab/components/SouthPane.jsx

```diff
@@ -92,6 +92,7 @@ export class SouthPane extends React.PureComponent {
     if (latestQuery) {
       if (
         isFeatureEnabled(FeatureFlag.SQLLAB_BACKEND_PERSISTENCE) &&
+        latestQuery.state === 'success' &&
```

Comment:
Nit: might want to create a enum or put the state in some centralized place instead of hard coding :). I have no idea what are the valid query states

Reply:
@hongbosong I totally agree, I had to look at another file in order to figure out this. I'll refactor this into an enum in a separate PR.